quantum/action_tapping.c

```diff
@@ -117,6 +117,75 @@ void action_tapping_process(keyrecord_t record) {
     }
 }
 
+/* Some conditionally defined helper macros to keep process_tapping more
+ * readable
+ */
+#    if (defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT)) || defined(TAPPING_TERM_PER_KEY) || defined(PERMISSIVE_HOLD_PER_KEY) || defined(TAPPING_FORCE_HOLD_PER_KEY) || defined(HOLD_ON_OTHER_KEY_PRESS_PER_KEY)
+#        define DEFINE_tapping_keycode uint16_t tapping_keycode = get_record_keycode(&tapping_key, false)
+#        define GET_TAP_TAPPING_TERM() GET_TAPPING_TERM(tapping_keycode, &tapping_key)
+#    else
+#        define DEFINE_tapping_keycode
+#        ifdef DYNAMIC_TAPPING_TERM_ENABLE
+#            define GET_TAP_TAPPING_TERM() g_tapping_term
+#        else
+#            define GET_TAP_TAPPING_TERM() TAPPING_TERM
+#        endif
+#    endif
+
+#    if defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT)
+#        ifdef RETRO_TAPPING_PER_KEY
+#            define GET_RETRO_TAPPING() get_retro_tapping(tapping_keycode, &tapping_key)
+#        else
+#            define GET_RETRO_TAPPING() true
+#        endif
+#        define IS_RETRO_SHIFTING(ev) \
+                 (GET_RETRO_TAPPING() && (RETRO_SHIFT + 0) != 0 && \
+                  TIMER_DIFF_16((ev).time, tapping_key.event.time) < (RETRO_SHIFT + 0))
+#        define RETROSHIFT_SWAP_TIMES() retroshift_swap_times()
+#        define IS_LT_TAPPING() IS_LT(tapping_keycode)
+#        define IS_MT_TAPPING() IS_MT(tapping_keycode)
+#        define IS_RETRO_TAPPING() IS_RETRO(tapping_keycode)
+#    else
+#        define GET_RETRO_TAPPING() false
+#        define IS_RETRO_SHIFTING(ev) false
+#        define RETROSHIFT_SWAP_TIMES()
+#        define IS_LT_TAPPING() false
+#        define IS_MT_TAPPING() false
+#        define IS_RETRO_TAPPING() false
+#    endif
+
+#    ifdef PERMISSIVE_HOLD_PER_KEY
+#        define GET_PERMISSIVE_HOLD() get_permissive_hold(tapping_keycode, &tapping_key)
+#    elif defined(PERMISSIVE_HOLD)
+#        define GET_PERMISSIVE_HOLD() true
+#    else
+#        define GET_PERMISSIVE_HOLD() false
+#    endif
+
+#    ifdef HOLD_ON_OTHER_KEY_PRESS_PER_KEY
+#        define GET_HOLD_ON_OTHER_KEY_PRESS() get_hold_on_other_key_press(tapping_keycode, &tapping_key)
+#    elif defined(HOLD_ON_OTHER_KEY_PRESS)
+#        define GET_HOLD_ON_OTHER_KEY_PRESS() true
+#    else
+#        define GET_HOLD_ON_OTHER_KEY_PRESS() false
+#    endif
+
+#    ifdef IGNORE_MOD_TAP_INTERRUPT_PER_KEY
+#        define GET_IGNORE_MOD_TAP_INTERRUPT() get_ignore_mod_tap_interrupt(tapping_keycode, &tapping_key)
+#    elif defined(IGNORE_MOD_TAP_INTERRUPT)
+#        define GET_IGNORE_MOD_TAP_INTERRUPT() true
+#    else
+#        define GET_IGNORE_MOD_TAP_INTERRUPT() false
+#    endif
+
+#    ifdef TAPPING_FORCE_HOLD_PER_KEY
+#        define GET_TAPPING_FORCE_HOLD() get_tapping_force_hold(tapping_keycode, &tapping_key)
+#    elif defined(TAPPING_FORCE_HOLD)
+#        define GET_TAPPING_FORCE_HOLD() true
+#    else
+#        define GET_TAPPING_FORCE_HOLD() false
+#    endif
+
```

Comment:
To stay consistent, all those macro functions should either take a `keycode` and `record` argument like the `GET_TAPPING_TERM` macro or transform the pre-existing `GET_TAPPING_TERM` into your `GET_TAP_TAPPING_TERM` macro. In which case, you could also drop all the `()` since they're unnecessary for a macro with no arguments.

Reply:
The macros don't have explicit arguments because the tapping_keycode variable may not be defined at all. I'm open to making the naming convention more consistent and dropping the ().